src/quantuminspire/calibration_viewer.py

```diff
@@ -0,0 +1,130 @@
+        """
+        Object to view and nicely print backend calibration data returned by the API.
+
+        :param calibration_dictionary: The calibration dictionary to be used by viewer.
+        """
+        self._calibration = calibration_dictionary
+        return
+
+    @property
+    def calibration(self) -> Dict[str, Any]:
+        """The property containing the calibration dictionary to be used."""
+        return self._calibration
+
+    @property
+    def timestamp(self) -> str:
+        """Timestamp of the last calibration of the current calibration dictionary."""
+        return str(self.calibration['parameters']['system']['last_calibration_date']['value'])
+
+    def show_system_parameters(self) -> None:
+        """Print system calibration parameters of the calibration dictionary."""
+        system = self.calibration['parameters']['system']
+        rprint(f'[green]System[/green]:')
+
+        for key, item in system.items():
+            if isinstance(item, dict):
+                if np.all([k in item for k in ['unit', 'value']]):
+                    rprint(f"\t{key}: {item['value']} [{item['unit']}]")
+                else:
+                    try:
+                        rprint(f"\t{key}: calibration at {item['timestamp']}")
+                    except KeyError:
+                        rprint(f"Key 'timestamp' not in item {item}.")
+            else:
+                rprint(f"Unknown structure '{key}' in system calibration.")
+
+    def show_qubit_parameters(self) -> None:
+        """Print qubit calibration parameters of the calibration dictionary."""
+        qdata = self.calibration['parameters']['qubits']
+
+        for qubit, data in qdata.items():
+            rprint(f'[green]Qubit[/green]: {qubit}')
+            for key, item in data.items():
+                if isinstance(item, dict):
+                    if np.all([k in item for k in ['unit', 'value']]):
+                        rprint(f"\t{key}: {item['value']} [{item['unit']}]")
+                    else:
+                        rprint(f"\t{key}: {item}")
+                else:
+                    rprint(f"Unknown structure '{key}' in qubit calibration.")
+
+    def get_calibration_field(self, field: str) -> Any:
+        """
+        Safely get value of key `field` from system calibration.
+
+        :param field: The entry (dict key) of the system calibration dictionary whose value should be returned.
+
+        :raises KeyError: If key `field` does not exist in system calibration dictionary.
+
+        :return: The system calibration content of `field`.
+        """
+        if field not in self.calibration['parameters']['system']:
+            raise KeyError(f"Calibration field '{field}' does not exist in system calibration dictionary.")
+
+        return self.calibration['parameters']['system'][field]
+
+    def show_calibration_field(self, field: str) -> Any:
+        """
+        Print calibration for a specific key `field` and return the content.
```

Comment:
This is what I expect but it doesn't do this. It doesn't print the field when 'value' key is part of the structure.